chain/stmgr/stmgr.go

```diff
@@ -55,12 +57,37 @@ func (sm *StateManager) TipSetState(ctx context.Context, ts *types.TipSet) (cid.
 
 	ck := cidsToKey(ts.Cids())
 	sm.stlk.Lock()
+	cw, cwok := sm.compWait[ck]
+	if cwok {
+		sm.stlk.Unlock()
+		span.AddAttributes(trace.BoolAttribute("waited", true))
+		select {
+		case <-cw:
+			sm.stlk.Lock()
+		case <-ctx.Done():
+			return cid.Undef, cid.Undef, ctx.Err()
+		}
+	}
 	cached, ok := sm.stCache[ck]
-	sm.stlk.Unlock()
 	if ok {
+		sm.stlk.Unlock()
 		span.AddAttributes(trace.BoolAttribute("cache", true))
 		return cached[0], cached[1], nil
 	}
+	ch := make(chan struct{})
```

Comment:
Only small nit. would be making this name longer `chWait`, so it is easier to spot in defer lambda bellow.

Reply:
This is from a golang guidelines I've read, the longer some value is going to be referenced by something, the longer it name should be (so it is easier to spot and memorize).

Reply:
I wouldn't block on that

Reply:
That is why I said it was only a nit.